### ISSUE TITLE
Use available task actions to determine whether we can create a child task

### ISSUE THREAD

**Issue by lowellrex** [department-of-veterans-affairs/caseflow#8273]
Before we create `Task` objects we confirm that the person attempting to create the task can do so (by way of `Task.verify_user_can_create!` and the overridden methods in its various subclasses). The only way users create tasks is through the available task actions dropdown on the case details page [1], which we populate through with the `Task.available_actions_unwrapper` method. Since we limit the set of actions that appear in that dropdown to only those actions that the current user can take, we can use the presence of a task in that set of actions to determine if the task we are trying to create can be created without including additional logic in the task models.

Using `MailTask` as an example, [`verify_user_can_create!`](https://github.com/department-of-veterans-affairs/caseflow/blob/afb9e9aa636a18da8bd1cc12a0c504139796e6e1/app/models/tasks/mail_task.rb#L22) (I'm using the new names for these functions after #8248 is merged) limits the set of folks who can create `MailTask`s to members of the `MailTeam`. The only place `MailTask`s are made available to be created is [in `RootTask.available_actions`](https://github.com/department-of-veterans-affairs/caseflow/blob/afb9e9aa636a18da8bd1cc12a0c504139796e6e1/app/models/tasks/root_task.rb#L11) which only displays the option to create a `MailTask` if the current user is a member of the `MailTeam`. Instead of duplicating this logic we could use the existence of `MailTask` in the parent task's set of available actions to determine whether the current user can create the task. Perhaps something like:

```ruby
def create_from_params(params, user)
  parent = Task.find(params[:parent_id])
  verify_user_can_create!(parent, user)
  ...

def verify_user_can_create!(parent, user)
  parent.available_actions_unwrapper(user).any? do |opt|
    opt.data && opt.data.type && opt.data.type == type
  end
end
```

## Benefits of this approach
* Less code in our codebase.
* Fewer paths through our code.
* Eliminates logic that effectively accomplishes the same thing twice.
* Allows us to make a change to who can create a task in a single place, reducing the likelihood that, for example, we want to allow another group of people to create `MailTask`s so we add them to the condition in `RootTask.available_actions` but forget to change `MailTask.verify_user_can_create!`.

## Caveats
* We will need a fallback for when we create `ColocatedTask`s from `AttorneyLegacyTask`s because those tasks will not have parent tasks.
* We will need to include the type of task we are creating in each option we put into the array of available actions. Currently the shape of this array looks like:
```ruby
[
  {
    label: "Mark task complete",
    value: "modal/mark_task_complete"
  },
  {
    label: "Human-readable label",
    value: "URL path", 
    data: { # Optional field
      selected: "value of currently selected option in dropdown",
      options: [ { label: "label", value: 12 }, ... ],
      type: GenericTask.name
    },
  },
  ...
]
```
* It may make sense to consider alternative approaches building this set of available actions (maybe moving from anonymous hashes to objects for the individual options and the `available_actions_unwrapper`?) if that makes it easier to access the information we want.
* We have a [`before_create`](https://github.com/department-of-veterans-affairs/caseflow/blob/afb9e9aa636a18da8bd1cc12a0c504139796e6e1/app/models/task.rb#L12) hook that sets the `assigned_at` value of the task and changes the parent task's status to `on_hold`. We will need to split this into two different hooks (maybe `before_create :set_assigned_at` and `after_create :put_parent_task_on_hold`).

## Acceptance criteria
* [ ] `Task.verify_user_can_create!` determines if somebody can create a task based on the parent task's `available_actions` instead of bespoke `verify_user_can_create!` methods for each task subclass.

[1] Tasks can also be created with the judge bulk assign functionality and automatically by the application in a handful of places. But the automatic task creation will bypass `Task.create_from_params` and any task that is judge bulk assign-able will have the "Assign to attorney" option as one of the available actions for that task.

**Comment by amprokop** [comment]
This sounds good to me. Are we planning on migrating colocated to use organizational queues in the medium term? Continuing to build on the three different ways we use tasks (judge/attorney, colocated, organization) *feels* like the risk here to me. You'd know better though.

**Comment by lowellrex** [comment]
Yup! #8119 and #8277 might be the only things standing in the way of us and a single queue table view for all individual peoples (probably not extending to organizations just yet).
